lib/utils/storage.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:shared_preferences/shared_preferences.dart';
+
+import '../constants.dart';
+import '../strings.dart';
+
+/// Persistent Storage on device read/write
+/// Read Settings values at app startup and set app state for these values
+/// (app state continue to use Provider package), These app state values are
+/// used multiple places through the code for each setting. This is approach
+/// is preferred rather than read the value from persistent storage at multiple
+/// places throughout the code. This minimises the reads/writes from/to
```

Comment:
This isn't actually correct: on Android the SharedPrefs framework code already does in-memory read and write caching so no need attempt to replicate that in Flutter side. I think iOS equivalent may also do this.

Reply:
Ok didn't know that, but the point is still valid either way, should use Provider rather than lots of calls to SharedPreferences all over the code. Also as per comment, setting up for possible other future settings.

Reply:
Oh complete agree on using Provider and abstracting shared prefs access 💯 on that. My feedback just on the read/write comment